File: src/models/Ingredient.php
<?php

namespace models;

class Ingredient
{
    private int $recipe_id;
    private string $amount;
    private string $product;
    public function __construct(int $recipe_id, string $amount, string $product){
        $this->recipe_id = $recipe_id;
        $this->amount = $amount;
        $this->product = $product;
    }
    public function getRecipeId(): int
    {
        return $this->recipe_id;
    }

    public function setRecipeId(int $recipe_id): void
    {
        $this->recipe_id = $recipe_id;
    }

    public function getAmount(): string
    {
        return $this->amount;
    }

    public function setAmount(string $amount): void
    {
        $this->amount = $amount;
    }

    public function getProduct(): string
    {
        return $this->product;
    }

    public function setProduct(string $product): void
    {
        $this->product = $product;
    }



}

File: OneDrive/Pulpit/demo/demo/src/main/java/com/example/demo/model/Ingredient.java
package com.example.demo.model;
import jakarta.persistence.*;

@Entity
@Table(name="ingredients")
public class Ingredient {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "product", nullable = false)
    private String product;
    @Column(name = "amount", nullable = false)
    private String amount;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "recipe_id", nullable = false)
    private Recipe recipe;

    public Ingredient() {}

    public Ingredient(String product, String amount, Recipe recipe) {
        this.product = product;
        this.amount = amount;
        this.recipe = recipe;
    }

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getProduct() { return product; }
    public void setProduct(String product) { this.product = product; }
    public String getAmount() { return amount; }
    public void setAmount(String amount) { this.amount = amount; }
    public Recipe getRecipe() {
        return recipe;
    }

    public void setRecipe(Recipe recipe) {
        this.recipe = recipe;
    }
}
